### DISCUSSION TITLE
Can I route db_nmap over a route

### DISCUSSION THREAD

**Discussion by flamecopper** [rapid7/metasploit-framework#16241] [Q&A]
Hi there, I would much preferred to route db_nmap over a route rather than using auxillary in place.
This is after setting the following command,

**run post/multi/manage/autoroute OPTION=s 
run autoroute -p**

But realised the the host that were accessible via route I can access via nmap db.
Is there a tweak that can make this work or is development in place?

**Comment by smcintyre-r7** [comment]
No you can't unfortunately. Forwarding over Metasploit's internal routing table is handled by our Rex::Socket subsystem. That's used by (almost) all framework components and Ruby modules (not external module). Since `db_nmap` runs the external `nmap` binary which uses it's own socket subsystem, it won't utilize Metasploit's routing capabilities.

You could start the socks proxy module and then run nmap through that, but you'd need to start the module yourself and then set the necessary options in nmap to run it through the proxy and of course things like syn scans won't work.

**Comment by its0x08** [comment]
You can scan separately using `nmap` outside of the metasploit and then just import the xml output of `nmap` to the **metasploit** using `db_import`.
